### ISSUE TITLE
pypi release github action for python sdk

### ISSUE THREAD

**Issue by asaiacai** [kubernetes-sigs/jobset#736]
**What would you like to be added**:

Nightly builds of the python sdk to pypi

**Why is this needed**:

It'd be easier to install from pypi.

**Comment by kannon92** [comment]
To be clear, installing of the python-sdk would not mean that jobset is "installed" on your cluster. The SDK provides bindings to create jobsets from python. But if a cluster does not have the CRD then the sdk would fail.

I think we could consider pypi wheels on a release of jobset. I don't think nightly builds would be worth the squeeze at the moment.

cc @ahg-g @danielvegamyhre

**Comment by asaiacai** [comment]
i see. yeah nightly might be early, but i agree that pypi release wheels would already be helpful in this situation.

**Comment by radoslawc** [comment]
Hi @asaiacai and @kannon92 is this still needed? If so I'd like to volunteer myself to work on that.

**Comment by kannon92** [comment]
@radoslawc sorry for the late response.

This is all yours. If you need help, I'd suggest asking on #sigs-k8s-infra on k8s slack. I am not sure how python publishing works in the Kubernetes org but I'm sure it has been figured out somewhere.

**Comment by rahulmansharamani14** [comment]
Hi @kannon92 @asaiacai , is someone working on it? If not, I'd love to investigate further and find the solution to this.

**Comment by kannon92** [comment]
I haven't heard anything.

**Comment by andreyvelich** [comment]
Thanks for your time @rahulmansharamani14!
It would be really nice if we could publish the JobSet API modules to PyPI similar to what we do for Kubeflow Trainer: https://github.com/kubeflow/trainer/tree/master/api/python_api
Here is the script to generate API Modules: https://github.com/kubeflow/trainer/blob/master/hack/python-api/gen-api.sh

In that case, other users can re-use the JobSet API in their SDKs, like we do in the Kubeflow SDK: https://github.com/kubeflow/sdk/blob/main/pyproject.toml#L32

**Comment by rahulmansharamani14** [comment]
Thanks @andreyvelich @kannon92  for the resources. After a quick review of Kubeflow Trainer, I have a plan to:

1. Set up auto-generation of Python API bindings from JobSet CRDs (similar to Kubeflow's approach)
2. Create a dedicated PyPI package for the API models
3. Add GitHub Actions workflow for automated publishing

Would it be helpful if I opened a draft PR with the proposed structure and implementation plan for feedback?

**Comment by andreyvelich** [comment]
Hi @rahulmansharamani14, that sounds good, please check the director structure and scripts we use for Trainer, we can do the same for JobSet. The package name can be: `jobset_api`

Additionally, you can remove the entire SDK directory from JobSet, as it is no longer required:
https://github.com/kubernetes-sigs/jobset/tree/main/sdk/python

**Comment by kannon92** [comment]
> Add GitHub Actions workflow for automated publishing

I don't know what options we have in kubernetes org on publishing python packages. It may be worth asking this in k8s slack .

We do have a kubernetes-python client so maybe we should see how they publish and follow that.

**Comment by andreyvelich** [comment]
Good point, @rahulmansharamani14 please can you start 🧵 in the `#kubernetes-client` Slack channel: https://kubernetes.slack.com/archives/C76GB48RK

**Comment by rahulmansharamani14** [comment]
Hi @andreyvelich @kannon92, apologies for the delay, got occupied with a life event. 
Is this something I can still work on? I can pick it up right away? Thanks!
